xray/manager.go

```diff
@@ -134,22 +135,22 @@ func (sm *XrayServicesManager) GetScanGraphResults(scanID string, includeVulnera
 }
 
 // BuildScan scans a published build-info with Xray.
-// Returns a string represents the scan ID.
-func (sm *XrayServicesManager) BuildScan(params services.XrayBuildParams) (*services.BuildScanResponse, error) {
+// 'scanResponse' - Xray scan output of the requested build scan.
+// 'noFailBuildPolicy' - indicated that Scan api got "No Xray Fail build...." error
+func (sm *XrayServicesManager) BuildScan(params services.XrayBuildParams, includeVulnerabilities bool) (scanResponse *services.BuildScanResponse, noFailBuildPolicy bool, err error) {
 	buildScanService := services.NewBuildScanService(sm.client)
 	buildScanService.XrayDetails = sm.config.GetServiceDetails()
-	err := buildScanService.Scan(params)
+	err = buildScanService.Scan(params)
 	if err != nil {
-		return nil, err
+		// if includeVulnerabilities flag is true and error is "No Xray Fail build...." continue to GetBuildScanResults to get vulnerabilities
```

Comment:
if includeVulnerabilities flag... --> If the includeVulnerabilities flag...
...and error is... --> ...and the error received is...